### DISCUSSION TITLE
Tuplas

### DISCUSSION THREAD

**Discussion by leonelsanchesdasilva** [DesignLiquido/delegua#617] [Ideias]
Hoje tá temos suporte a tuplas para o dialeto Potigol, mas não para Delégua. Em Potigol, tuplas são definidas da seguinte forma:

```pascal
var t := (1, 2)
```

Pela especificação de Potigol, podemos ter tuplas de até 10 elementos. Cada tipo de tupla é nomeado da seguinte forma:

- 2 elementos: dupla
- 3 elementos: trio
- 4 elementos: quarteto
- 5 elementos: quinteto
- 6 elementos: sexteto
- 7 elementos: septeto
- 8 elementos: octeto
- 9 elementos: noneto
- 10 elementos: deceto

Inicialmente, podemos implementar o suporte a tuplas em Delégua igual é implementado em Potigol, e depois divergindo futuramente se isso for uma boa ideia.

Minha ideia seria usar uma combinação de símbolos para definir o literal de uma tupla, visto que usar parênteses (como em Python e Potigol) e colchetes (como em JavaScript, que se combina com vetores) é problemático por aumentar deveras a complexidade da avaliação sintática. 

Algumas ideias:

```js
var t = [(1, 2)]
var t = \(1, 2)\
var t = |(1, 2)|
```

**Comment by JohnPetros** [comment]
Haveria também uma função para transformar iteráveis em tuplas? Similar ao Python: 

```
lista = [1,2,3]
print(tuple(lista))
```

**Reply by leonelsanchesdasilva** [reply]
@JohnPetros Com certeza. Ótima ideia.

**Reply by leonelsanchesdasilva** [reply]
https://github.com/DesignLiquido/delegua/issues/629

**Comment by JohnPetros** [comment]
Acho a notação de `[(1, 2)]` mais interessante.

**Comment by samuelrvg** [comment]
Resolvido na PR https://github.com/DesignLiquido/delegua/pull/627

Versão [0.32.0](https://github.com/DesignLiquido/delegua/releases/tag/0.32.0).
